### ISSUE TITLE
Bug: I get "TypeError: Illegal invocation" when using node-fetch@3 with electron

### ISSUE THREAD

**Issue by badersur** [node-fetch/node-fetch#917]
I'm not sure if this is a bug but I think so. I have an electron app that uses node-fetch@2 which is working fine (in a GUI and the terminal), but with version 3 when I only run the app in the terminal (I get a TypeError when run in the GUI).

This is maybe due to dropping support for the browser but I don't think so! The app uses node-fetch in the GUI using a preload script which have access to full node APIs as far as I understand, and it should be running as if it's running from a terminal! Or maybe it's an issue with [Electron](https://github.com/electron/electron/issues/21517) itself?!

**Reproduction**

Using node >=12.14.1 should be fine but I use version 14.8.0.

1. `git clone https://github.com/badersur/miniapp.git`
2. `cd miniapp`
3. `git checkout node-fetch-3`
4. `yarn`
5. `yarn dev`

**Expected behavior**
The app should work fine as shown in the master branch:

```
git checkout master
yarn
yarn dev
# Passing `-t` to the app will output to the terminal (which is working fine using both versions), for Linux I use:
yarn dist:dir && ./dist/linux-unpacked/miniapp -t
```

**Screenshots**

![Screenshot from 2020-08-18 18-11-01](https://user-images.githubusercontent.com/1686391/90523672-3801ea80-e17e-11ea-8f53-41eca671dc5f.png)

**Your Environment**

| software         | version
| ---------------- | -------
| node-fetch       | 3.0.0-beta.8
| node             | 14.8.0
| npm              | 6.14.7
| yarn              | 1.22.4
| Operating System | Ubuntu 20.04.1 LTS

**Additional context**

**Comment by TimoZikeli** [comment]
Have you found some workaround for this bug? Thanks!

**Comment by badersur** [comment]
No, but I tried other libraries (`got`, `axios` & `superagent`) that worked so I'm thinking it's a bug with node-fetch not Electron!

I'm still wondering why `got` worked even though they say that it have *no* support for Electron & the browser!

**Comment by jimmywarting** [comment]
`Illegal invocation` normally happens when a function gets unaware/detached from it's class.

example:
![Skärmavbild 2020-09-08 kl  20 07 19](https://user-images.githubusercontent.com/1148376/92512359-f6111500-f20e-11ea-9cc5-6fc684ca03e0.png)

This can easily be fixing if you bind the function `set = header.set.bind(header)`

I see that there is a `proxy` is involved... It could be the cause of something failing? maybe it calls any of the sub methods with the wrong context when eg doing `obj.apply(ctx, arguments)` or calling anything like that.

It seems to be something wrong with this line:

https://github.com/node-fetch/node-fetch/blob/38839c53bd417cef440757265f75c8371d4c51e6/src/headers.js#L128-L131

is `receiver` undefined or anything like that?

---

wondering why we even use proxy and urlsearchparams, it don't look nice to mix it into the Header class to begin with.

**Comment by xxczaki** [comment]
cc @tinovyatkin

**Comment by JosephusPaye** [comment]
Just ran into the same issue. Is there anything we can do to help fix the issue?

**Comment by MiniSuperDev** [comment]
Same here
